Vcsim error while creating datastore

Getting error while starting vcsim with default parameters:

```
$ ./vcsim

panic: runtime error: index out of range
goroutine 1 [running]:
panic(0xb82140, 0xc42000c120)
        /usr/lib/golang/src/runtime/panic.go:500 +0x1a1
github.com/vmware/govmomi/simulator.(*Datastore).RefreshDatastore(0xc420092f00, 0xc4202174a0, 0x0, 0x0)
        /root/.go/src/github.com/vmware/govmomi/simulator/datastore.go:81 +0x4c6
github.com/vmware/govmomi/simulator.(*HostDatastoreSystem).CreateLocalDatastore(0xc420088f50, 0xc4201e8fc0, 0x0, 0x0)
        /root/.go/src/github.com/vmware/govmomi/simulator/host_datastore_system.go:117 +0x437
reflect.Value.call(0xc13e20, 0xc420088f50, 0x213, 0xd3a541, 0x4, 0xc4201a47b0, 0x1, 0x1, 0xc13e20, 0xaea060, ...)
        /usr/lib/golang/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0xc13e20, 0xc420088f50, 0x213, 0xc4201a47b0, 0x1, 0x1, 0xc420088f50, 0x213, 0x20)
        /usr/lib/golang/src/reflect/value.go:302 +0xa4
github.com/vmware/govmomi/simulator.(*Service).call(0xc4200cad20, 0xc4201a48e0, 0x196, 0xaea060)
        /root/.go/src/github.com/vmware/govmomi/simulator/simulator.go:144 +0x359
github.com/vmware/govmomi/simulator.(*Service).RoundTrip(0xc4200cad20, 0x12aa600, 0xc42005c338, 0x12a10c0, 0xc4202171c0, 0x12a10c0, 0xc4202171e0, 0x1, 0xc4202171e0)
        /root/.go/src/github.com/vmware/govmomi/simulator/simulator.go:168 +0x2d2
github.com/vmware/govmomi/vim25.(*Client).RoundTrip(0xc4200ce2c0, 0x12aa600, 0xc42005c338, 0x12a10c0, 0xc4202171c0, 0x12a10c0, 0xc4202171e0, 0x40, 0xc87ba0)
        /root/.go/src/github.com/vmware/govmomi/vim25/client.go:66 +0x80
github.com/vmware/govmomi/vim25/methods.CreateLocalDatastore(0x12aa600, 0xc42005c338, 0x12a0b00, 0xc4200ce2c0, 0xc4201e8fc0, 0x8b7119, 0xd15160, 0xc4201e8f80)
        /root/.go/src/github.com/vmware/govmomi/vim25/methods/methods.go:2559 +0xe7
github.com/vmware/govmomi/object.HostDatastoreSystem.CreateLocalDatastore(0x0, 0x0, 0xc4200ce2c0, 0xa885bf, 0x13, 0xc4201c8420, 0x16, 0x12aa600, 0xc42005c338, 0xc4201da7f0, ...)
        /root/.go/src/github.com/vmware/govmomi/object/host_datastore_system.go:44 +0x1ba
github.com/vmware/govmomi/simulator.(*Model).createLocalDatastore(0xc420196500, 0xc4201da7f0, 0x9, 0xc4201d4b00, 0x4, 0x4, 0xa, 0x12aa600)
        /root/.go/src/github.com/vmware/govmomi/simulator/model.go:455 +0x380
github.com/vmware/govmomi/simulator.(*Model).Create(0xc420196500, 0xd3e741, 0xc)
        /root/.go/src/github.com/vmware/govmomi/simulator/model.go:420 +0x22a5
main.main()
        /root/.go/src/github.com/vmware/govmomi/vcsim/main.go:76 +0x7dd
```

Starting vcsim with no datastores and VM's (-ds 0 -vm 0) it starts with no errors. Using govc I checked that it did create all the other default objects. Using govc command to create datastore I got the following error:

```
$./vcsim -vm 0 -ds 0
GOVC_URL=https://user:pass@127.0.0.1:8989/sdk
2017/07/12 08:35:55 http: panic serving 127.0.0.1:53854: runtime error: index out of range
goroutine 16 [running]:
net/http.(*conn).serve.func1(0xc420428100)
        /usr/lib/golang/src/net/http/server.go:1491 +0x12a
panic(0xb82140, 0xc420010110)
        /usr/lib/golang/src/runtime/panic.go:458 +0x243
github.com/vmware/govmomi/simulator.(*Datastore).RefreshDatastore(0xc42008d400, 0xc4201fc260, 0x0, 0x0)
        /root/.go/src/github.com/vmware/govmomi/simulator/datastore.go:81 +0x4c6
github.com/vmware/govmomi/simulator.(*HostDatastoreSystem).CreateLocalDatastore(0xc4202fdb30, 0xc4201f3840, 0x0, 0x0)
        /root/.go/src/github.com/vmware/govmomi/simulator/host_datastore_system.go:117 +0x437
reflect.Value.call(0xc13e20, 0xc4202fdb30, 0x213, 0xd3a541, 0x4, 0xc4200fda28, 0x1, 0x1, 0xc13e20, 0xaea060, ...)
        /usr/lib/golang/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0xc13e20, 0xc4202fdb30, 0x213, 0xc4200fda28, 0x1, 0x1, 0xc4202fdb30, 0x213, 0xc4201f3b00)
        /usr/lib/golang/src/reflect/value.go:302 +0xa4
github.com/vmware/govmomi/simulator.(*Service).call(0xc4200c2ec0, 0xc4201f3b00, 0x600, 0xc4201f3b00)
        /root/.go/src/github.com/vmware/govmomi/simulator/simulator.go:144 +0x359
github.com/vmware/govmomi/simulator.(*Service).ServeSDK(0xc4200c2ec0, 0x12a9f00, 0xc4201a36c0, 0xc4203191d0)
        /root/.go/src/github.com/vmware/govmomi/simulator/simulator.go:261 +0xb01
github.com/vmware/govmomi/simulator.(*Service).ServeSDK-fm(0x12a9f00, 0xc4201a36c0, 0xc4203191d0)
        /root/.go/src/github.com/vmware/govmomi/simulator/simulator.go:389 +0x48
net/http.HandlerFunc.ServeHTTP(0xc420011560, 0x12a9f00, 0xc4201a36c0, 0xc4203191d0)
        /usr/lib/golang/src/net/http/server.go:1726 +0x44
net/http.(*ServeMux).ServeHTTP(0xc420188780, 0x12a9f00, 0xc4201a36c0, 0xc4203191d0)
        /usr/lib/golang/src/net/http/server.go:2022 +0x7f
net/http.serverHandler.ServeHTTP(0xc420064380, 0x12a9f00, 0xc4201a36c0, 0xc4203191d0)
        /usr/lib/golang/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc420428100, 0x12aa5c0, 0xc4203ae980)
        /usr/lib/golang/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
        /usr/lib/golang/src/net/http/server.go:2293 +0x44d
```
This does create govcsim temp folders in /tmp/, but they are empty
I am using CentOS 6.8 as root user. Using the same commands as I did when setting this up on Windows (where it worked)

--- a/simulator/datastore.go
+++ b/simulator/datastore.go
@@ -17,9 +17,6 @@ limitations under the License.
 package simulator
 
 import (
-	"os/exec"
-	"strconv"
-	"strings"
 	"time"
 
 	"github.com/vmware/govmomi/object"
@@ -46,27 +43,13 @@ func parseDatastorePath(dsPath string) (*object.DatastorePath, types.BaseMethodF
 func (ds *Datastore) RefreshDatastore(*types.RefreshDatastore) soap.HasFault {
 	r := &methods.RefreshDatastoreBody{}
 
-	info := ds.Info.GetDatastoreInfo()
-
-	// #nosec: Subprocess launching with variable
-	buf, err := exec.Command("df", "-k", info.Url).Output()
-
+	err := ds.stat()
 	if err != nil {
 		r.Fault_ = Fault(err.Error(), &types.HostConfigFault{})
 		return r
 	}
 
-	lines := strings.Split(string(buf), "\n")
-	columns := strings.Fields(lines[1])
-
-	used, _ := strconv.ParseInt(columns[2], 10, 64)
-	info.FreeSpace, _ = strconv.ParseInt(columns[3], 10, 64)
-
-	info.FreeSpace *= 1024
-	used *= 1024
-
-	ds.Summary.FreeSpace = info.FreeSpace
-	ds.Summary.Capacity = info.FreeSpace + used
+	info := ds.Info.GetDatastoreInfo()
 
 	now := time.Now()
 

--- a/simulator/os_unix.go
+++ b/simulator/os_unix.go
@@ -1,0 +1,38 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simulator
+
+import "syscall"
+
+func (ds *Datastore) stat() error {
+	info := ds.Info.GetDatastoreInfo()
+	var stat syscall.Statfs_t
+
+	err := syscall.Statfs(info.Url, &stat)
+	if err != nil {
+		return err
+	}
+
+	bsize := uint64(stat.Bsize) / 512
+
+	info.FreeSpace = int64(stat.Bfree*bsize) >> 1
+
+	ds.Summary.FreeSpace = info.FreeSpace
+	ds.Summary.Capacity = int64(stat.Blocks*bsize) >> 1
+
+	return nil
+}

--- a/simulator/os_windows.go
+++ b/simulator/os_windows.go
@@ -1,0 +1,26 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package simulator
+
+import "os"
+
+func (ds *Datastore) stat() error {
+	info := ds.Info.GetDatastoreInfo()
+
+	_, err := os.Stat(info.Url)
+	return err
+}

